hedera-node/src/main/java/com/hedera/services/state/virtual/UniqueTokenValue.java

```diff
@@ -0,0 +1,211 @@
+package com.hedera.services.state.virtual;
+
+import com.google.common.base.MoreObjects;
+import com.hedera.services.state.merkle.MerkleUniqueToken;
+import com.hedera.services.state.submerkle.EntityId;
+import com.hedera.services.state.submerkle.RichInstant;
+import com.swirlds.common.io.SerializableDataInputStream;
+import com.swirlds.common.io.SerializableDataOutputStream;
+import com.swirlds.jasperdb.files.DataFileCommon;
+import com.swirlds.virtualmap.VirtualValue;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Objects;
+
+import static com.hedera.services.state.merkle.internals.BitPackUtils.*;
+import static java.lang.Math.min;
+
+/**
+ * Represents the information stored in the virtualized merkle node associated with a unique token (NFT).
+ */
+public class UniqueTokenValue implements VirtualValue {
+	private static final long CLASS_ID = 0xefa8762aa03ce697L;
+	// Maximum amount of metadata bytes allowed.
+	private static final int MAX_METADATA_BYTES = 100;
+	/** Current version of the encoding scheme. */
+	/* package */ static final int CURRENT_VERSION = 1;
+
+	/** The account number field of the owner's account id. */
+	private long ownerAccountNum;
+	/**
+	 * The compressed creation time of the token:
+	 *  - the higher 32-bits represents an unsigned integer value containing the number of seconds since the epoch
+	 *    representing the consensus time at which the token was created.
+	 *  - the lower 32-bits represents a signed integer containing nanosecond resolution to the timestamp in which the
+	 *    token was created.
+	 */
+	private long packedCreationTime;
+	/** The metadata associated with the unique token, the maximum number of bytes for this field is 100 bytes. */
+	private byte[] metadata = new byte[0];
+	/** Whether this instance is immutable (e.g. from fast copy operation). */
+	private boolean isImmutable = false;
```

Comment:
NFTs are always immutable.

Reply:
According to the FastCopyable interface (under #isImmutable), the most recent copy must be mutable. So, not sure how to make this instance truly immutable when the FastCopyable interface says otherwise, and de-serialization requires the fields to be settable.

Reply:
And looks like it isn't truly immutable. The owners can change, so the concept of mutability still exists. Currently, I have no setters, but this will change when I do the integration and need to set the owner field.